src/test/java/net/snowflake/client/jdbc/CustomProxyIT.java

```diff
@@ -31,6 +32,56 @@
 public class CustomProxyIT {
   @Rule public TemporaryFolder tmpFolder = new TemporaryFolder();
 
+  /**
+   * Before running this test, change the user and password to appropriate values. Set up 2
+   * different proxy ports that can run simultaneously. This is easy with Burpsuite.
+   *
+   * <p>This tests that separate, successful connections can be made with 2 separate proxies at the
+   * same time.
+   *
+   * @throws SQLException
+   */
+  @Test
+  @Ignore
+  public void test2Proxies() throws SQLException {
```

Comment:
Is it possible to test the httpClientWithProxies directly? We just need to make sure the same types of connections is reused.